datahub-web-react/src/app/entity/domain/DataProductsTab/EditDataProductModal.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button, Modal, message } from 'antd';
+import React, { useState } from 'react';
+import DataProductBuilderForm from './DataProductBuilderForm';
+import { DataProductBuilderState } from './types';
+import { useUpdateDataProductMutation } from '../../../../graphql/dataProduct.generated';
+import { DataProduct } from '../../../../types.generated';
+import { MODAL_BODY_STYLE, MODAL_WIDTH } from './CreateDataProductModal';
+
+type Props = {
+    dataProduct: DataProduct;
+    onClose: () => void;
+    onUpdateDataProduct: (dataProduct: DataProduct) => void;
+};
+
+export default function EditDataProductModal({ dataProduct, onUpdateDataProduct, onClose }: Props) {
```

Comment:
Do we need separate modal for create and edit?

Reply:
hm probably not. right now they contain their own mutations separately and reuse the same `DataProductBuilderForm` component so there isn't much duplication. Can refactor if desired otherwise I think it's alright (could do in a followup as well)